Test/DurableTask.AzureStorage.Tests/TestTablePartitionManager.cs

```diff
@@ -0,0 +1,723 @@
+            {
+                TaskHubName = testName,
+                StorageConnectionString = this.connection,
+                WorkerId = "0",
+                AppName = testName,
+                UseTablePartitionManagement = true,
+            };
+            var service = new AzureStorageOrchestrationService(settings);
+            await service.StartAsync();
+
+
+            WaitForCondition(
+                timeout: TimeSpan.FromSeconds(5),
+                condition: () =>
+                {
+                    var partitions = service.ListTableLeases();
+                    Assert.AreEqual(4, partitions.Count());
+                    return partitions.All(p => p.CurrentOwner == "0");
+                });
+
+            await service.StopAsync();
+            await service.DeleteAsync();
+        }
+
+        //Starts with two workers and four partitions.
+        //Test that one worker can acquire two partitions. 
+        //Since two worker can not start at the same time, and one will start earlier than the another one.
+        //There would be a steal process, and test that the lease tranfer will take no longer than 30 sec.
+        [TestMethod]
+        public async Task TestTwoWorkerWithFourPartitions()
+        {
+            var services = new AzureStorageOrchestrationService[2];
+            string testName = nameof(TestTwoWorkerWithFourPartitions);
+
+            for (int i = 0; i < 2; i++)
+            {
+                var settings = new AzureStorageOrchestrationServiceSettings
+                {
+                    TaskHubName = testName,
+                    StorageConnectionString = this.connection,
+                    AppName = testName,
+                    UseTablePartitionManagement = true,
+                    WorkerId = i.ToString(),
+                };
+                services[i] = new AzureStorageOrchestrationService(settings);
+            }
+
+            var startTasks = services.Select(service => service.StartAsync());
+            await Task.WhenAll(startTasks);
+
+            WaitForCondition(
+                timeout: TimeSpan.FromSeconds(30),
+                condition: () =>
+                {
+                    var partitions = services[0].ListTableLeases();
+                    Assert.AreEqual(4, partitions.Count());
+                    return ((partitions.Count(p => p.CurrentOwner == "0") == 2) &&
+                            (partitions.Count(p => p.CurrentOwner == "1") == 2));
+                });
+
+            var stopTasks = services.Select(service => service.StopAsync());
+            await Task.WhenAll(stopTasks);
+            await services[0].DeleteAsync();
+        }
+
+        //Starts with four workers and four partitions.
+        //Test that each worker can acquire four partitions. 
+        //Since workers can not start at the same time, there would be a steal lease process.
+        //Test that the lease tranfer will take no longer than 30 sec.
```

Comment:
nit: let's add a space between `//` and the first character of each line
also some minor correction - this test is checking that each worker can acquire 1 partition, not 4
```suggestion
        // Starts with four workers and four partitions.
        // Test that each worker can acquire one partition. 
        // Since workers can't start at the same time, there should be a steal lease process.
        // Test that the lease transfer will take no longer than 30 sec.
```